### PR TITLE
RUMM-299 Prepare project for UI integration testing

### PR DESCRIPTION
**Pull request by ncreated**
### What and why?

With new features coming, the old targets setup in `Datadog.xcworkspace` is not enough.

Old targets:
* `Datadog` - framework target with SDK Swift code;
* `DatadogObjc` - framework target with SDK Swift bridge to Objective-C;
* `DatadogTests` - unit tests target with SKD tests;
* `DatadogIntegrationTests` - unit tests target with integration tests and benchmarks using python server mock.

This layout has few limitations:
* impossible to run unit tests and benchmarks on a real device (because of no host app target);
* impossible to interact with actual UI in integration tests (crucial for upcoming RUM feature);
* depending on SDK internals for integration tests (`@testable import`);
* forcing the same setup for integration tests and benchmarks in single target.

🚚 📦 This PR updates the `Datadog.xcworkspace` layout and solves all above issues ✨ .

### How?

New targets were introduced and existing were updated:

Screenshot 2020-05-06 at 09 29 46

* `Datadog` - framework target.
* `DatadogObjc` - framework target.
* `DatadogTests` - unit tests target. Now uses **host app**, so it can be run also on a real device.
* `DatadogBenchmarkTests` - **unit tests target** using host app, to run tests on a real device.
* `DatadogIntegrationTests` - **UI tests target** with integration tests using python mock server.
* `Example` - host app target.

Separating different kind of tests enables different setup for each. Important differences between `Datadog*` test targets are:

| Target                      | PBX.productType  | Configuration  | Instrumentation      |
| ------------------  |:------------------:| --------------:| -------------------:|
| `Tests`                     | `.unit-test`           | `Debug`          | Thread Sanitizer      |
| `BenchmarkTests ` | `.unit-test`           | `Integration`   | no |
| `IntegrationTests `  | `.ui-testing`         | `Release`        | `HTTPServerMock` |

This mimics the production environment as much as we can, with providing the maximum qualification of the SDK code. Integration tests use `Release` configuration, unit tests leverage the [Thread Sanitizer](https://developer.apple.com/documentation/code_diagnostics/thread_sanitizer) and benchmarks run fast with no extra instrumentation overhead (`Integration` configuration is a copy of `Release` with enabled testability).

By segregating tests by their nature instead of setup-similarities, we also enable them to any further extensions or more instrumentation if needed.

#### The `Example` app target
The main purpose of this target is to lay close with SDK code within the same `Datadog.xcworkspace`, so it can be easily updated with new features coming to the SDK. Both "Logging" and "Tracing" debug screens were moved to this target, leaving the `Shopist` project clean so it can become an ultimate demonstration of the SDK API and features (no littering with "technical hacks" for SDK debugging purpose).

Now, by simply launching the `Example` scheme we can send data directly to Datadog and browse it in Logs and Traces explorers. The same `Example` target, when launched by any of test schemes behaves differently:

| Scheme + command.       | UI                                                  | Server url  | Purpose  |
| ------------------------  |:-------------------------------:| --------------:| -------------------:|
| `Example`, `CMD`+`R`   | Loging & Tracing debug menu  | `datadoghq.com` | Sending & debugging real data on `app.datadoghq.com` |
|  `Datadog`, `CMD`+`U`  | no UI, `window = nil`                 | - | Acting as the host target for unit tests
| `DatadogBenchmarkTests`, `CMD` + `U`   | no UI, `window = nil`                 | - | Acting as the host target for benchmarks
| `DatadogIntegrationTests`, `CMD` + `U`   | Logging & Tracing fixtures menu | python mock server url | Acting as a real app navigated by UITest runner

#### 🎁 Other things done in this PR

New targets required code signing setup for running them on physical device. As the `xcconfig` setup in the repo became messy, I refactored it to only use two `*.xcconfig` files stored in the root folder:

Screenshot 2020-05-06 at 10 28 14

Both contain only two things: `DATADOG_CLIENT_TOKEN` and `DEVELOPMENT_TEAM`. The `*.local.xcconfig` is git-ignored, so we can store our secrets in there. All the Xcode project targets in this repository were updated to read values from `Datadog.xcconfig`. This includes `Shopist.xcodeproj`, `Datadog.xcworkspace` and 3 projects under `dependency-manager-tests`.

#### ⚠️ Not done in this PR

To separate code-changes for easier code review, this PR contains only infrastructure changes + necessary code updates. In next #98, purely code-related PR I will:
* 1/ update integration tests to send data from `Example` app;
* 2/ add Tracing integration tests;
* 3/ change `DatadogIntegrationTests's` configuration from `Integration` to `Release` (not yet possible in this PR due to 1).

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -52,12 +52,8 @@ $ make
 
 #### Tests
 
-`DatadogTests` (unit tests) and `DatadogIntegrationTests` source files
+`DatadogTests` (unit tests) and `DatadogIntegrationTests` (integration tests) and `DatadogBenchmarkTests` (benchmarks) source files
```
Comment (ruthnaebeck) [inline CONTRIBUTING.md:55]:
```suggestion
`DatadogTests` (unit tests), `DatadogIntegrationTests` (integration tests), and `DatadogBenchmarkTests` (benchmarks) source files
```

Comment (ncreated) [reply CONTRIBUTING.md:55]: Good catch, thank you @ruthnaebeck 👌

File: bitrise.yml [hunk 2]
```diff
@@ -101,7 +112,7 @@ workflows:
         - content: |-
             #!/usr/bin/env bash
             set -e
-            cd Shopist && make
+            echo 'DATADOG_CLIENT_TOKEN="fake-client-token-for-shopist"' > xcconfigs/Datadog.local.xcconfig
```
Comment (buranmert) [inline bitrise.yml:115]: ℹ️ Makefile for Shopist can take `client_token` as an argument, `make Shopist secret="fake-client-token"` should work as well

Comment (ncreated) [reply bitrise.yml:115]: Yay, saw it, but I wanted to limit number of `xcconfig` files, so the `./Shopist/shopist-secrets.local.xcconfig` was removed in this PR (together with Shopist's `Makefile`). Now all projects in the repository read the config from `xcconfigs/Datadog.local.xcconfig`.

**Pull request merged**